website_sale_preset_carts/models/res_partner.py

```diff
@@ -48,3 +52,21 @@ def signup_retrieve_info(self, token):
         res['nb_household'] = partner.nb_household
         res['subscriptions'] = self.sudo().get_subscriptions()
         return res
+
+    @api.multi
+    def write(self, vals):
+        for partner in self:
+            if 'cart_suspended_from' in vals and 'cart_suspended_date' in vals:
+                suspended_from = vals['cart_suspended_from']
+                suspended_to = vals['cart_suspended_date']
+            elif 'cart_suspended_from' in vals:
+                suspended_from = vals['cart_suspended_from']
+                suspended_to = partner.cart_suspended_date
+            elif 'cart_suspended_date' in vals:
+                suspended_from = partner.cart_suspended_from
+                suspended_to = vals['cart_suspended_date']
+            else:
+                return super(ResPartner, partner).write(vals)
+            if suspended_from > suspended_to:
+                raise UserError(_("Cart Suspended from date can't be after Cart Suspended Until date."))
+            return super(ResPartner, partner).write(vals)
```

Comment:
Shouldn't this be done under a `@api.contrains` function? I also don't like that there is a return and call to super statement in several point of the function.

Reply:
nice catch !